### ISSUE TITLE
After updating Rocket.Chat to 3.4.0 some clients (app or browser) won't switch to Login-form

### ISSUE THREAD

**Issue by ggruber4711** [RocketChat/Rocket.Chat#18120]
### Description:
Some clients will not work after update of Rocket.Chat from 3.0.3 to 3.4.0.
If you open the rocket.chat URL directly in the browser, you would only see the white box, but the redirection to the login form would not occur.

### Steps to reproduce:

No clear reproduction given.
It happens just on some clients. It seems to be network related. As soon as this issue occurs, the rocket.chat client would not work correctly neither on any browser on that machine nor on the windows app.

### Server Setup Information:

- Version of Rocket.Chat Server:  3.4.0
- Operating System:   Linux
- Deployment Method: docker
- Number of Running Instances:  1
- DB Replicaset Oplog:  Enabled
- NodeJS Version:  12.16.1
- MongoDB Version:  4.0.16

### Client Setup Information

- Desktop App or Browser Version:  Latest Chrome, Firefox, etc.
- Operating System:  Windows 10

### Relevant logs:

Screenshot Browser:
![image](https://user-images.githubusercontent.com/3330414/86338303-771dc000-bc52-11ea-8952-4a1dc04823df.png)

Screenshot of browser console errors:
![image](https://user-images.githubusercontent.com/3330414/86338203-548ba700-bc52-11ea-8719-9d494c402ea6.png)

Javascript Errors in the browser console:

```
d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48 POST https://chat.workflow.at/api/v1/method.callAnon/public-settings%2Fget 404 (Not Found)
send @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48
ajax @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48
l @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
_jqueryCall @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
post @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
post @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
e @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
sendMessage @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
apply @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
call @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
a.call @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
r @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
loadFromServer @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
loadFromServerAndPopulate @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
init @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
async function (async)
init @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
L @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
get @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
i @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
e @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
e @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
Show 21 more frames
d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413 Error: Not Found
    at originalCallback (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411)
    at Object.processTwoFactorError (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411)
    at Object.a [as error] (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411)
    at c (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)
    at Object.fireWith [as rejectWith] (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)
    at w (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)
    at XMLHttpRequest.<anonymous> (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:1
Promise.then (async)
n.then @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:105
e @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
sendMessage @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
apply @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
call @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
a.call @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
r @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
loadFromServer @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
loadFromServerAndPopulate @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
init @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
async function (async)
init @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
L @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
get @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
i @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
e @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
o @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:5
t @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
g @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
n @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
e @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:3
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
Show 18 more frames
d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48 POST https://chat.workflow.at/api/v1/method.callAnon/public-settings%2Fget 404 (Not Found)
send @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48
ajax @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48
l @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
_jqueryCall @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
post @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
post @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411
e @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413
sendMessage @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
_sendOutstandingMethods @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
_callOnReconnectAndSendAppropriateOutstandingMethods @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
onReset @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:147
(anonymous) @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
forEachCallback @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
_connected @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
socket.onopen @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
r.dispatchEvent @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
k._dispatchOpen @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
k._didMessage @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
o.ws.onmessage @ d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:145
d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:413 Error: Not Found
    at originalCallback (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411)
    at Object.processTwoFactorError (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411)
    at Object.a [as error] (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:411)
    at c (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)
    at Object.fireWith [as rejectWith] (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)
    at w (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)
    at XMLHttpRequest.<anonymous> (d0ca345489ba3b05cc9dc310dd15b659dd17ea28.js?meteor_js_resource=true:48)

```

**Comment by Dave-zl** [comment]
+1 We have the same problem after upgrade to 3.4.0

**Comment by aleadco** [comment]
#18128

**Comment by aleadco** [comment]
`Jul 02 12:45:42 chat rocketchat[3263]:     'x-forwarded-proto': 'https',
Jul 02 12:45:42 chat rocketchat[3263]:     host: 'chat.cneuro.cu',
Jul 02 12:45:42 chat rocketchat[3263]:     'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, l
Jul 02 12:45:42 chat rocketchat[3263]:     'accept-language': 'es-419,es;q=0.9,es-ES;q=0.8,en;q=0.7,en-GB;q=0.6,en-US;q=0.5'
Jul 02 12:45:42 chat rocketchat[3263]:   },
Jul 02 12:45:42 chat rocketchat[3263]:   userId: null
Jul 02 12:45:42 chat rocketchat[3263]: }
Jul 02 12:45:42 chat rocketchat[3263]: server.js:204 Meteor ➔ publish null -> userId: null, arguments: []`

**Comment by serviceman** [comment]
+1 having the exact shame issue upgrading from version 3.0.12

**Comment by stylefish** [comment]
i'm facing this issue too, this is related to the "public-settings/get" and a reverse proxy (in my case apache):

when rocket.chat tries to resolve this path: `api/v1/method.callAnon/public-settings%2Fget` it get a `404 (Not Found)` because apache does not like the url encoded slash in the request (%2F) i think this problem is related to that.

**Comment by stylefish** [comment]
fixed for me in 3.4.2

**Comment by vitamindck** [comment]
broken in 3.5.0 docker image for me
